library/core/src/intrinsics.rs

```diff
@@ -959,13 +959,14 @@ extern "rust-intrinsic" {
     #[rustc_safe_intrinsic]
     pub fn assert_zero_valid<T>();
 
-    /// A guard for unsafe functions that cannot ever be executed if `T` has invalid
-    /// bit patterns: This will statically either panic, or do nothing.
+    /// A guard for `std::mem::uninitialized`. Checks whether a repeated bit pattern `0x01`
```

Comment:
> Checks whether a repeated bit pattern `0x01` is legal for `T`

Except if we decide to change the bit pattern (unlikely) or `-Zstrict-init-checks` is passed. Yes it's redundant, but it might be better to just say "Checks whether `mem::uninitialized` should panic for `T`" 🤷